### PR TITLE
refactor(TokenInput): remove flag

### PR DESCRIPTION
**Pull request by fernandomg**
# Description:

Uses `singleToken` prop as a receiver of the required Token. This way we avoid conditionally checking things (`singleToken` vs `token`).

Also,  the `useTokenSearch` API was refactored, so it better mimics hooks API in general with the dependency array as the second parameter.

## Type of change:

- [ ] New feature
- [ ] Bug fix
- [ ] Breaking change
- [ ] Enhancement
- [x] Refactoring
- [ ] Chore

# How Has This Been Tested?

- [x] Manual testing
- [ ] Automated tests
- [ ] Other (explain)

# Remember to check that:

- Your code follows the style guidelines of this project
- You have performed a self-review of your code
- You have commented your code in hard-to-understand areas
- You have made corresponding changes to the documentation
- Your changes generate no new warnings

### CODE REVIEW

File: src/components/sharedComponents/TokenInput/index.tsx [hunk 1]
```diff
@@ -47,12 +47,11 @@ export const CloseButton = styled(BaseCloseButton)`
 `
 
 interface TokenInputProps extends Omit<TokenSelectProps, 'onError'> {
-  singleToken?: boolean
-  token?: Token
   onAmountSet: (amount?: string) => void
   onError: (error?: string) => void
   thousandSeparator?: boolean
   title?: string
+  singleToken?: Token
```
Comment (nicosampler) [inline src/components/sharedComponents/TokenInput/index.tsx:54]: Doing this change causes it to lose one of the possible configurations.

1. If `singleToken == false` and no `token` is provided: any token is pre-selected, and the user can select any token from any chain. 
2. if `singleToken ==  false` and a `token` is provided: the select initializes with a pre-selected token, and the user can freely change chain and token. 
3. if `singleToken == true` and `token` is provided: the chain and token are fixed. The user cannot change anything related to the token.

4. I just realized we should do something similar with the chain, sometimes I would want to fix the chain so the user can't change it.

File: src/hooks/useTokenSearch.ts [hunk 2]
```diff
@@ -19,14 +24,15 @@ type TokenSearch = {
  * A hook that provides a performant search to filter a list of tokens by a searchTerm
  * Internally it uses React's `useDeferredValue`
  *
- * @param {Array} tokens - a list of tokens to be filtered by `searchTerm`
- * @param {Array | undefined} deps - array of dependencies that trigger recalculation of the search
+ * @param {object} options - options object
+ * @param {string} [options.defaultSearchTerm] - the default search term
```
Comment (nicosampler) [inline src/hooks/useTokenSearch.ts:28]:
```suggestion
 * @param {string} [options.defaultSearchTerm] - the default search term used to find a partial match against address, symbol, and  name
```

File: src/components/sharedComponents/TokenInput/index.tsx (hunk 1 above)
Comment (fernandomg) [reply src/components/sharedComponents/TokenInput/index.tsx:54]: oh, oh... 🤦 nevermind then. Let's leave it as it is

Comment (fernandomg) [reply src/components/sharedComponents/TokenInput/index.tsx]: @nicosampler I'd rolled back the changes and was looking at the chains implementation.

If you pass an array of 1 element to 'network' you fix the number of chains, and the user won't be able to change it.

Please, check if that works for you.

image

**Comment by fernandomg** [comment]
@nicosampler I'm looking for an approval. If you happen to have one, please send it over here 🙏

**Review by nicosampler — APPROVED** [review]

**Pull request merged**